packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1718,6 +1787,9 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             if (this.gcTestMode) {
                 this.dataStores.deleteUnusedRoutes(deletedNodeIds);
             }
+
+            // Update the GC data as per this run.
+            this.gcDataFromLastRun = cloneGCData(gcData);
```

Comment:
This should not be updated here. This should only happen when we get the ack for the summary, similar to how we track other start across summaries.

Reply:
So, what's next? :) I.e. are you planning to do this in this PR, or open an issue for future, or something else?

Reply:
I should have probably mentioned that :-). This is a pain every time we have something new to track. I am looking at how to simplify this. Maybe have a separate class like `RootSummarizerNode` that tracks all of this stuff for container runtime. Or maybe we can simplify this tracking process. But yeah, that has to happen before this PR can go in.
So, I will update this either in this PR or create a separate PR for this.